seeq/addons/correlation/utils/_cache_management.py

```diff
@@ -0,0 +1,70 @@
+import sys
+import os
+from setuptools import find_packages
+from pkgutil import iter_modules
+import seeq.addons.correlation as correlation
+
+
+def _clear_cache(foo):
+    """
+    Clears the cache of only one function
+
+    Parameters
+    ----------
+    foo: {obj},
+        Function object whose cache will be clear
+
+    Returns
+    -------
+    None
+    """
+    try:
+        foo.cache_clear()
+        print("Cache of function '{}' has been cleared".format(foo.__name__))
+    except AttributeError:
+        raise AttributeError('Function "{}" does not utilize cache. Nothing to clear'.format(foo.__name__))
+    return
+
+
+def clear_cache_all():
+    """
+    Inspects the seeq.addons.causality module and clears the cache of all objects
+
+    """
+    modules = find_all_modules(correlation)
+    package_foos = find_all_functions(modules, correlation)
+    for foo in package_foos:
+        try:
+            foo.cache_clear()
+            print("Cache of function '{}' has been cleared".format(foo.__name__))
+        except AttributeError:
+            pass
+
+
+def find_all_modules(package_obj):
```

Comment:
You might want to limit this to only loaded submodules since you are using it in the clear cache and importing all of them might not be desired. The following code worked for me for that purpose:
```python
import inspect
def find_loaded_submodules(module):
    def is_not_standard_module(m):
        return inspect.ismodule(m) and m.__package__
    
    def is_submodule_of_module(m):
        return module.__package__ in m.__package__ and m.__package__ != module.__package__
    
    g = globals().copy()  # globals() will be updating during iteration, so take a snapshot  
    for k, v in g.items():
        if is_not_standard_module(v) and is_submodule_of_module(v):
            print(f"{v.__name__} is a submodule of {module.__name__}")
```
Give the output
```python
>>> import seeq
>>> find_loaded_submodules(seeq)

>>> from seeq import spy
>>> find_loaded_submodules(seeq)
seeq.spy is a submodule of seeq

>>> from seeq import sdk
>>> find_loaded_submodules(seeq)
seeq.spy is a submodule of seeq
seeq.sdk is a submodule of seeq
```

Reply:
@sethgilchrist The intent here is to find all the functions under `seeq.addons.correlation` and clear cache only for those functions. If I understand correctly, the code you are sharing would get the loaded submodules one of which is `seeq.addons.correlation`. However, I'm already manually picking only that submodule for cache clearing. Am I missing something?

Maybe the name of my functions is misleading too. I named it `find_all_modules` thinking on the python jargon that a module is a .py file.